bond/bond.go

```diff
@@ -351,30 +351,35 @@ func cmdAdd(args *skel.CmdArgs) error {
 		return err
 	}
 
+	// Initialize an L2 default result.
 	result := &current.Result{
 		CNIVersion: cniVersion,
+		Interfaces: []*current.Interface{bondInterface},
 	}
+
 	// run the IPAM plugin and get back the config to apply
 	if bondConf.IPAM.Type != "" {
 		r, err := ipam.ExecAdd(bondConf.IPAM.Type, args.StdinData)
 		if err != nil {
 			return err
 		}
 		// Convert whatever the IPAM result was into the current Result type
-		result, err := current.NewResultFromResult(r)
+		ipamResult, err := current.NewResultFromResult(r)
 		if err != nil {
 			return err
 		}
 
-		if len(result.IPs) == 0 {
+		if len(ipamResult.IPs) == 0 {
 			return errors.New("IPAM plugin returned missing IP config")
 		}
-		for _, ipc := range result.IPs {
+		for _, ipc := range ipamResult.IPs {
 			// All addresses belong to the vlan interface
 			ipc.Interface = current.Int(0)
 		}
 
-		result.Interfaces = []*current.Interface{bondInterface}
+		result.IPs = ipamResult.IPs
+		result.Routes = ipamResult.Routes
+		result.DNS = ipamResult.DNS
```

Comment:
this one is being overwritten below, in line 391 by whatever sits at `bondConf.DNS` . 

Which one of those should take precedence ?...

I recommend leaving this attribute out of the fray, focus on the issue at hand (missing IPs / routes), and discuss what should be done regarding DNS in a follow up PR / issue.